conflict between micrometer and stork

### Describe the bug

When I add in the same application the stork extension and the micrometer extension I get an error.

### Expected behavior

Should not conflict with each other.

### Actual behavior

When I build the application I get:

```
[INFO] Running org.acme.ReactiveGreetingResourceTest
2022-02-11 18:56:35,295 INFO  [org.jbo.threads] (main) JBoss Threads version 3.4.2.Final
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.273 s <<< FAILURE! - in org.acme.ReactiveGreetingResourceTest
[ERROR] org.acme.ReactiveGreetingResourceTest.testHelloEndpoint  Time elapsed: 0.007 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: javax.enterprise.inject.CreationException: 
Synthetic bean instance for io.quarkus.micrometer.runtime.binder.HttpBinderConfiguration not initialized yet: io_quarkus_micrometer_runtime_binder_HttpBinderConfiguration_97d170e1550eee4afc0af065b78cda302a97674c
        - a synthetic bean initialized during RUNTIME_INIT must not be accessed during STATIC_INIT
        - RUNTIME_INIT build steps that require access to synthetic beans initialized during RUNTIME_INIT should consume the SyntheticBeansRuntimeInitBuildItem
```

### How to Reproduce?

Get https://github.com/vsevel/stork-issue/tree/micrometer_stork_conflict
```
mvn clean package
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/smallrye-stork/deployment/pom.xml
+++ b/extensions/smallrye-stork/deployment/pom.xml
@@ -21,6 +21,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-deployment</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/smallrye-stork/deployment/src/main/java/io/quarkus/stork/deployment/SmallRyeStorkProcessor.java
+++ b/extensions/smallrye-stork/deployment/src/main/java/io/quarkus/stork/deployment/SmallRyeStorkProcessor.java
@@ -2,6 +2,7 @@ package io.quarkus.stork.deployment;
 
 import static java.util.Arrays.asList;
 
+import io.quarkus.arc.deployment.SyntheticBeansRuntimeInitBuildItem;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.Consume;
@@ -30,6 +31,7 @@ public class SmallRyeStorkProcessor {
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     @Consume(RuntimeConfigSetupCompleteBuildItem.class)
+    @Consume(SyntheticBeansRuntimeInitBuildItem.class)
     void initializeStork(SmallRyeStorkRecorder storkRecorder, ShutdownContextBuildItem shutdown) {
         storkRecorder.initialize(shutdown);
     }

--- a/extensions/smallrye-stork/runtime/pom.xml
+++ b/extensions/smallrye-stork/runtime/pom.xml
@@ -22,6 +22,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

